ProcessLib/Utils/SetOrGetIntegrationPointData.h

```diff
@@ -0,0 +1,132 @@
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    auto const num_intpts = ip_data.size();
+
+    cache.clear();
+    auto cache_mat = MathLib::createZeroedMatrix<Eigen::Matrix<
+        double, kelvin_vector_size, Eigen::Dynamic, Eigen::RowMajor>>(
+        cache, kelvin_vector_size, num_intpts);
+
+    for (unsigned ip = 0; ip < num_intpts; ++ip)
+    {
+        auto const& sigma = ip_data[ip].sigma_eff;
+        cache_mat.col(ip) =
+            MathLib::KelvinVector::kelvinVectorToSymmetricTensor(sigma);
+    }
+
+    return cache;
+}
+
+template <typename IP_Data, int DisplacementDim>
+std::vector<double> const& getIntPtEpsilon(IP_Data const& ip_data,
+                                           std::vector<double>& cache)
+{
+    constexpr int kelvin_vector_size =
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    auto const num_intpts = ip_data.size();
+
+    cache.clear();
+    auto cache_mat = MathLib::createZeroedMatrix<Eigen::Matrix<
+        double, kelvin_vector_size, Eigen::Dynamic, Eigen::RowMajor>>(
+        cache, kelvin_vector_size, num_intpts);
+
+    for (unsigned ip = 0; ip < num_intpts; ++ip)
+    {
+        auto const& eps = ip_data[ip].eps;
+        cache_mat.col(ip) =
+            MathLib::KelvinVector::kelvinVectorToSymmetricTensor(eps);
+    }
+
+    return cache;
+}
+
+template <typename IP_Data, int DisplacementDim>
+std::vector<double> getSigma(IP_Data const& ip_data)
+{
+    constexpr int kelvin_vector_size =
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    auto const num_intpts = ip_data.size();
+
+    std::vector<double> ip_sigma_values;
+    auto cache_mat = MathLib::createZeroedMatrix<Eigen::Matrix<
+        double, Eigen::Dynamic, kelvin_vector_size, Eigen::RowMajor>>(
+        ip_sigma_values, num_intpts, kelvin_vector_size);
+
+    for (unsigned ip = 0; ip < num_intpts; ++ip)
+    {
+        auto const& sigma = ip_data[ip].sigma_eff;
+        cache_mat.row(ip) =
+            MathLib::KelvinVector::kelvinVectorToSymmetricTensor(sigma);
+    }
+
+    return ip_sigma_values;
+}
+
+template <typename IP_Data, int DisplacementDim>
+std::size_t setSigma(double const* values, IP_Data& ip_data)
+{
+    constexpr int kelvin_vector_size =
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    auto const num_intpts = ip_data.size();
+
+    auto sigma_values =
+        Eigen::Map<Eigen::Matrix<double, kelvin_vector_size, Eigen::Dynamic,
+                                 Eigen::ColMajor> const>(
+            values, kelvin_vector_size, num_intpts);
+
+    for (unsigned ip = 0; ip < num_intpts; ++ip)
+    {
+        ip_data[ip].sigma_eff =
+            MathLib::KelvinVector::symmetricTensorToKelvinVector(
+                sigma_values.col(ip));
+    }
+
+    return num_intpts;
+}
+
+template <typename IP_Data, typename IP_DataMember>
+std::vector<double> const& getIntegrationPintScalarData(
```

Comment:
This function may increase a little bit compilation time.

Reply:
How much?

Reply:
Typo

Reply:
That was just an estimation before. Actually there is no change in the build time due to this PR:
**Comparison the times of  full building in release on my laptop:**
```
ufz/master:        ninja -j4  4293.27s user 97.02s system 394% cpu 18:31.89 total
Branch of this PR: ninja -j4  4288.14s user 97.04s system 394% cpu 18:31.43 total
```